src/posix/http_client_curl.cc

```diff
@@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
   FILE* pipe = ::popen(cmd.c_str(), "r");
   assert(pipe);
   char s[8 * 1024];
-  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
+  std::string result;
+  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
+    result += s;
```

Comment:
Didn’t get your example. It works correctly (e.g. fully reads all lines from the file, including the last one, which doesn’t end with \n):
- 4 chars for first line (“foo\n”)
- 3 chars for the last one (“bar”)

What is wrong then?

> On Mar 15, 2015, at 00:55, Dima <notifications@github.com> wrote:
> 
> In src/posix/http_client_curl.cc <https://github.com/KnowSheet/Alohalytics/pull/27#discussion_r26442142>:
> 
> > @@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
> >    FILE* pipe = ::popen(cmd.c_str(), "r");
> >    assert(pipe);
> >    char s[8 * 1024];
> > -  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
> > +  std::string result;
> > +  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
> > +    result += s;
> Yes:
> 
> dima@dima-i9:~/fgets$ cat 1.cc 
> #include <cstdio>
> #include <cstring>
> 
> int main() {
>   FILE* f;
>   f = fopen("dima_test.txt", "w");
>   fprintf(f, "foo\nbar");
>   fclose(f);
>   f = fopen("dima_test.txt", "r");
>   char s[1024];
>   while (fgets(s, sizeof(s), f)) {
>     printf("%d\n", strlen(s));
>   }
>   fclose(f);
> }
> dima@dima-i9:~/fgets$ g++ 1.cc 
> dima@dima-i9:~/fgets$ ./a.out 
> 4
> 3
> —
> Reply to this email directly or view it on GitHub <https://github.com/KnowSheet/Alohalytics/pull/27/files#r26442142>.
>